src/util/artifacts.js

```diff
@@ -1,25 +1,54 @@
 /* @flow */
-import {fs} from 'mz';
+import {fs as defaultFS} from 'mz';
+import mkdirp from 'mkdirp';
+import promisify from 'es6-promisify';
 
 import {UsageError, isErrorWithCode} from '../errors';
 import {createLogger} from './logger';
 
 const log = createLogger(__filename);
-
+const promisifiedMkdirp = promisify(mkdirp);
 
 export async function prepareArtifactsDir(
-  artifactsDir: string
+  artifactsDir: string,
+  fs?: typeof defaultFS = defaultFS,
```

Comment:
Let's remove the injected `fs` and `mdir` dependency and let's try to use the tmpDir based testing approach describe in the related review comment.

(it will make this change smaller and the tests without sinon will make the change even safer)